lib/ansible/modules/clustering/consul_acl.py

```diff
@@ -151,56 +158,48 @@ def execute(module):
 
 
 def update_acl(module):
-
     rules = module.params.get('rules')
-    state = module.params.get('state')
     token = module.params.get('token')
     token_type = module.params.get('token_type')
     mgmt = module.params.get('mgmt_token')
     name = module.params.get('name')
     consul = get_consul_api(module, mgmt)
     changed = False
 
-    try:
+    rules = decode_rules_as_yml(rules)
+    rules_as_hcl = encode_rules_as_hcl_string(rules) if len(rules) > 0 else None
 
+    try:
         if token:
             existing_rules = load_rules_for_token(module, consul, token)
-            supplied_rules = yml_to_rules(module, rules)
-            changed = not existing_rules == supplied_rules
+            changed = existing_rules != rules
```

Comment:
The reason it passed [the tests](https://github.com/ansible/ansible/blob/devel/test/integration/roles/test_consul_acl/tasks/main.yml) was because they were rather minimial and did not test for idempotence ;).

I spent some time yesterday [improving them](https://github.com/wtsi-hgi/ansible/tree/feature/consul_acl_rules/test/integration/roles/test_consul_acl/tasks) and the issue you describe *should* now cause them to fail.

However, for some reason the [CI testing seems to be passing just fine](https://app.shippable.com/github/ansible/ansible/runs/26238/summary/console)? If I run the `test_consul_acl` independently on the Ansible test setup, they fail as expected though. If anyone can spot anything wrong with the way these tests are setup, please let me know.